openvdb_houdini/SOP_NodeVDB.cc

```diff
@@ -711,10 +711,24 @@ OpenVDBOpFactory::OpenVDBOpFactory(
     setNativeName(OpenVDBOpPolicy().getNativeName(*this));
 }
 
-OpenVDBOpFactory::~OpenVDBOpFactory()
+OpenVDBOpFactory&
+OpenVDBOpFactory::setNativeName(const std::string& name)
 {
-    if (!mNativeName.empty()) {
+    // SideFX nodes have no native equivalent.
+#ifndef SESI_OPENVDB
+    addSpareData({{"nativename", name}});
+
+    // if native name was previously defined and is present
+    // in the hidden table then remove it regardless of policy
+
+    if (!mNativeName.empty() &&
+        this->table().isOpHidden(mNativeName.c_str())) {
+        this->table().delOpHidden(mNativeName.c_str());
+    }
+
+    mNativeName = name;
 
+    if (!name.empty()) {
         // environment variable takes precedence over compiler flag
 
         const char* opHidePolicy = std::getenv("OPENVDB_OPHIDE_POLICY");
```

Comment:
You should `#include <cctype>` for `std::tolower`.

Also, how about moving all of the string manipulation logic (lines 734–749) into a function?
```
namespace {

const std::string&
getOpHidePolicy()
{
    static bool sInitialized = false;
    static std::string sOpHidePolicy;
    if (!sInitialized) {
        const char* opHidePolicy = std::getenv("OPENVDB_OPHIDE_POLICY");
        ...
        sOpHidePolicy = opHidePolicy;
        std::transform(sOpHidePolicy.begin(), sOpHidePolicy.end(), sOpHidePolicy.begin(),
            [](unsigned char c) { return std::tolower(c); });
        sInitialized = true;
    }
    return sOpHidePolicy;
}

} // anonymous namespace
```

Reply:
I've just implemented this (with a minor change to use std::call_once instead of a boolean).

Reply:
Doesn't the Mako comment belong with the `node_info_text` namespace?

Reply:
Not really, the mako comment belongs with this define that immediately follows it:

//#define OPENVDB_CUSTOM_MAKO

which refers to the block of code captured by #ifdef OPENVDB_CUSTOM_MAKO much later in the file (https://github.com/danrbailey/openvdb/blob/unifysops/openvdb_houdini/SOP_NodeVDB.cc#L283).

Reply:
Might be good at some point to rephrase that comment to something like "Define OPENVDB_CUSTOM_MAKO to enable custom UT_InfoTree data...".

Reply:
Yes, we could reword that a little so it's clearer.